main.tf

```diff
@@ -384,7 +384,7 @@ module "efs_csi_driver" {
   role_description              = try(var.efs_csi_driver.role_description, "IRSA for aws-efs-csi-driver project")
 
   role_policy_arns = try(var.efs_csi_driver.role_policy_arns,
-    { EfsCsiDriverPolicy = aws_iam_policy.efs_csi_driver[0].arn }
```

Comment:
We should use the generated policy, no? It will map the dependencies correctly and therefore the ordering